src/App.tsx

```diff
@@ -1,26 +1,125 @@
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [fetchedTodos, setFetchedTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterBy, setFilterBy] = useState<'all' | 'active' | 'completed'>(
+    'all',
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedTodoIds, setCompletedTodoIds] = useState<number[] | null>(
+    null,
+  );
+  const [isHiddenError, setIsHiddenError] = useState(true);
+
+  const timeoutRef = useRef<number | null>(null);
+
+  const removeErrorMessage = () => {
+    setIsHiddenError(true);
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+      timeoutRef.current = null;
+    }
+
+    timeoutRef.current = window.setTimeout(() => {
+      setErrorMessage('');
+    }, 1000);
+  };
+
+  const handleDelete = (id: number) => {
+    setTodos(prevTodo => prevTodo.filter(todo => todo.id !== id));
+  };
+
+  const errorMessageHandler = useCallback((er: Error) => {
+    setErrorMessage(er.message ?? String(er));
+    setIsHiddenError(false);
+
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
+    timeoutRef.current = window.setTimeout(() => {
+      removeErrorMessage();
+    }, 3000);
+  }, []);
+  // <br />
+  // Unable to update a todo
+
+  useEffect(() => {
+    getTodos()
+      .then(serverTodos => {
+        setTodos(serverTodos);
+        setFetchedTodos(serverTodos);
+      })
+      .catch(errorMessageHandler);
+  }, []);
+
+  useEffect(() => {
+    let filteredTodos = todos;
+
+    if (filterBy !== 'all') {
+      filteredTodos = todos.filter(todo =>
+        filterBy === 'active' ? !todo.completed : todo.completed,
+      );
+    }
+
+    setFetchedTodos([...filteredTodos]);
+  }, [filterBy, todos]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          onError={errorMessageHandler}
+          onSuccess={setTodos}
+          setTempTodo={setTempTodo}
+          errorMessage={errorMessage}
+        />
+
+        <TodoList
+          renderedList={fetchedTodos}
+          tempTodo={tempTodo}
+          todos={todos}
+          forClear={completedTodoIds}
+          setForClear={setCompletedTodoIds}
+          onDelete={setTodos}
```

Comment:
it looks strange
define delete function here and pass it (that might allow you not to pass `completedTodoIds` variable and handle clear completed here)

or at least change prop name